src/libexpr/eval-cache.cc

```diff
@@ -574,14 +574,15 @@ std::string AttrCursor::getString()
                 debug("using cached string attribute '%s'", getAttrPathStr());
                 return s->first;
             } else
-                root->state.error<TypeError>("'%s' is not a string", getAttrPathStr()).debugThrow();
+                // TODO: Is this an internal error? Where is literally any documentation for the eval cache?
```

Comment:
There probably is no documentation :(

From seeing how this is used, I think this cursor thing can either just traverse a regular "uncached" value, or traverse a cache entry's serialized value. The `getString` part is just the caller asking for a specific type; it's not an internal error.